frontend/front/src/pages/Admin/survey/SurveyProfile.js

```diff
@@ -1,152 +1,45 @@
-import React, { useState } from "react";
-import { Box, Button, TextField, Typography } from "@mui/material";
+import React, { useCallback, useMemo } from "react";
+import { Box, CircularProgress, Container, Typography } from "@mui/material";
 import { useParams } from "react-router-dom";
-import ConfirmationModal from "../../Developer/confirmModal/ConfirmationModal";
+import { useGetSurveyListQuery } from "../../../redux/apiSlice";
+import { SurveyError } from "./SurveyError";
+import { AdminSurvey } from "../../../components/SurveyComponent/SurveyComponent";
 
 const SurveyProfile = () => {
   const { uid } = useParams();
-  const [editMode, setEditMode] = useState(false);
-  const [deleteModal, setDeleteModal] = useState(false);
 
-  const confirmDelete = () => {
-    setDeleteModal(false);
-    // make api call to delete the user profile here
-  };
-  const cancelDelete = () => {
-    setDeleteModal(false);
-  };
+  const { survey, error, isLoading } = useGetSurveyListQuery(undefined, {
+    selectFromResult: ({ data }) => ({
+      // make sure ID is a string!
+      survey: data?.find((s) => `${s.id}` === uid),
+    }),
+  });
 
-  // Conditional Buttons
-  let changePasswordButton, editButton, deleteButton;
-  if (!editMode) {
-    changePasswordButton = (
-      <Button variant="contained" sx={{ ml: 2 }}>
-        CHANGE PASSWORD
-      </Button>
-    );
-    editButton = (
-      <Button
-        variant="outlined"
-        sx={{ ml: 2 }}
-        onClick={() => setEditMode(true)}
-      >
-        EDIT
-      </Button>
-    );
-    deleteButton = (
-      <Button
-        variant="outlined"
-        sx={{ ml: 2 }}
-        color="error"
-        onClick={() => setDeleteModal(true)}
-      >
-        DELETE
-      </Button>
-    );
-  } else {
-    editButton = (
-      <Button
-        variant="outlined"
-        sx={{ ml: 2 }}
-        onClick={() => setEditMode(false)}
-      >
-        SAVE
-      </Button>
-    );
-    deleteButton = (
-      <Button
-        variant="outlined"
-        sx={{ ml: 2 }}
-        color="error"
-        onClick={() => setEditMode(false)}
-      >
-        CANCEL
-      </Button>
+  const title = useMemo(() => `Survey ${uid}`, [uid]);
+
+  const PageContent = useCallback(() => {
+    if (survey) {
+      return <AdminSurvey defaultData={survey} />;
```

Comment:
I tried to reuse the existing survey component to display the survey data. I realize this is incomplete but I'm going to make a follow-up card to get all these surveys and the dummy data in line with the actual draft survey.